.github/workflows/sycl_macos_build_and_test.yml

```diff
@@ -0,0 +1,73 @@
+name: Reusable SYCL macOS build and test workflow
+
+on:
+  workflow_call:
+    inputs:
+      build_ref:
+        type: string
+        required: false
+      build_cache_suffix:
+        type: string
+        required: false
+        default: "default"
+      build_cache_size:
+        type: string
+        required: false
+        default: 2G
+      build_configure_extra_args:
+        type: string
+        required: false
+        default: ""
+      build_artifact_suffix:
+        type: string
+        required: false
+        default: "default"
+
+jobs:
+  build:
+    name: Build
+    runs-on: macos-12
+    steps:
+    - name: Install dependencies
+      run: brew install ccache ninja
+    - uses: actions/checkout@v3
```

Comment:
All other scripts are using v2, so this patch introduces inconsistency, which I'd like to avoid.
It would be great if you can update checkout version to v3 in all other places. Follow-up PR will work.

Reply:
Ok, I'll follow-up